--- a/vendor/cursive/src/backends/curses/pan.rs
+++ b/vendor/cursive/src/backends/curses/pan.rs
@@ -70,7 +70,7 @@
         pancurses::mouseinterval(0);
         pancurses::mousemask(
             pancurses::ALL_MOUSE_EVENTS | pancurses::REPORT_MOUSE_POSITION,
-            ::std::ptr::null_mut(),
+            None,
         );
 
         // This asks the terminal to provide us with mouse drag events
