--- a/rawk
+++ b/rawk
@@ -98,7 +98,7 @@
 
     if [ -n "$stylesheet" ]; then
         mkdir -p "$target"/styles
-        cp -u "$stylesheet" "$target"/styles/style.css
+        cp "$stylesheet" "$target"/styles/style.css
     fi
 
     cd "$src_dir"
@@ -119,7 +119,7 @@
             if [ -d "$i" ]; then
                 mkdir -p "$o"
             else
-                cp -u "$i" "$o"
+                cp "$i" "$o"
             fi
             ;;
         esac
