--- a/meson.build
+++ b/meson.build
@@ -306,7 +306,7 @@
 rt_lib = cc.find_library('rt', required : false) # clock_gettime
 dl_lib = cc.find_library('dl', required : false)
 pthread_lib = dependency('threads')
-dbus_dep = dependency('dbus-1')
+dbus_dep = dependency('dbus-1', required : false)
 sdl_dep = dependency('sdl2', required : get_option('sdl2'))
 ncurses_dep = dependency('ncursesw', required : false)
 sndfile_dep = dependency('sndfile', version : '>= 1.0.20', required : get_option('sndfile'))
--- a/src/modules/meson.build
+++ b/src/modules/meson.build
@@ -110,30 +110,30 @@
   dependencies : pipewire_module_protocol_deps,
 )
 
-pipewire_module_protocol_pulse_deps = pipewire_module_protocol_deps
-
 if dbus_dep.found()
+  pipewire_module_protocol_pulse_deps = pipewire_module_protocol_deps
+
   pipewire_module_protocol_pulse_deps += dbus_dep
-endif
 
-pipewire_module_protocol_pulse = shared_library('pipewire-module-protocol-pulse',
-  [ 'module-protocol-pulse.c',
-    'module-protocol-pulse/manager.c',
-    'module-protocol-pulse/pulse-server.c',
-    'module-protocol-pulse/modules/module-loopback.c',
-    'module-protocol-pulse/modules/module-native-protocol-tcp.c',
-    'module-protocol-pulse/modules/module-null-sink.c',
-    'module-protocol-pulse/modules/module-remap-sink.c',
-    'module-protocol-pulse/modules/module-remap-source.c',
-    'module-protocol-pulse/modules/module-simple-protocol-tcp.c',
-  ],
-  c_args : pipewire_module_c_args,
-  include_directories : [configinc, spa_inc],
-  install : true,
-  install_dir : modules_install_dir,
-  install_rpath: modules_install_dir,
-  dependencies : pipewire_module_protocol_pulse_deps,
-)
+  pipewire_module_protocol_pulse = shared_library('pipewire-module-protocol-pulse',
+    [ 'module-protocol-pulse.c',
+      'module-protocol-pulse/manager.c',
+      'module-protocol-pulse/pulse-server.c',
+      'module-protocol-pulse/modules/module-loopback.c',
+      'module-protocol-pulse/modules/module-native-protocol-tcp.c',
+      'module-protocol-pulse/modules/module-null-sink.c',
+      'module-protocol-pulse/modules/module-remap-sink.c',
+      'module-protocol-pulse/modules/module-remap-source.c',
+      'module-protocol-pulse/modules/module-simple-protocol-tcp.c',
+    ],
+    c_args : pipewire_module_c_args,
+    include_directories : [configinc, spa_inc],
+    install : true,
+    install_dir : modules_install_dir,
+    install_rpath: modules_install_dir,
+    dependencies : pipewire_module_protocol_pulse_deps,
+  )
+endif
 
 pipewire_module_protocol_simple = shared_library('pipewire-module-protocol-simple',
   [ 'module-protocol-simple.c' ],
--- a/spa/plugins/support/meson.build
+++ b/spa/plugins/support/meson.build
@@ -32,6 +32,7 @@
 		        install_dir : spa_plugindir / 'support')
 endif
 
+if dbus_dep.found()
 spa_dbus_sources = ['dbus.c']
 
 spa_dbus_lib = shared_library('spa-dbus',
@@ -40,7 +41,7 @@
 			dependencies : [dbus_dep, ],
 			install : true,
 		        install_dir : spa_plugindir / 'support')
-
+endif
 
 if systemd_dep.found()
   spa_journal_sources = ['journal.c']
