--- a/lib/route/cls/ematch_syntax.c
+++ b/lib/route/cls/ematch_syntax.c
@@ -218,12 +218,12 @@
 #define KW_CLOSE 265
 #define KW_PLUS 266
 #define KW_MASK 267
-#define mask 268
+// #define mask 268
 #define KW_SHIFT 269
 #define KW_AT 271
 #define at 272
 #define EMATCH_CMP 273
-#define cmp 274
+// #define cmp 274
 #define EMATCH_NBYTE 275
 #define pattern 276
 #define EMATCH_TEXT 277
--- a/lib/route/cls/ematch_syntax.h
+++ b/lib/route/cls/ematch_syntax.h
@@ -12,12 +12,12 @@
 #define KW_CLOSE 265
 #define KW_PLUS 266
 #define KW_MASK 267
-#define mask 268
+// #define mask 268
 #define KW_SHIFT 269
 #define KW_AT 271
 #define at 272
 #define EMATCH_CMP 273
-#define cmp 274
+// #define cmp 274
 #define EMATCH_NBYTE 275
 #define pattern 276
 #define EMATCH_TEXT 277
