diff --git a/library/unwind/build.rs b/library/unwind/build.rs
index 24bcd40c..51777664 100644
--- a/library/unwind/build.rs
+++ b/library/unwind/build.rs
@@ -19,7 +19,7 @@
     } else if target.contains("linux") {
         // linking for Linux is handled in lib.rs
         if target.contains("musl") {
-            llvm_libunwind::compile();
+            println!("cargo:rustc-link-lib=unwind");
         } else if target.contains("android") {
             let build = cc::Build::new();
diff --git a/library/unwind/src/lib.rs b/library/unwind/src/lib.rs
index dbdefa47..dca1897b 100644
--- a/library/unwind/src/lib.rs
+++ b/library/unwind/src/lib.rs
@@ -39,7 +39,7 @@ cfg_if::cfg_if! {
 
 #[cfg(target_env = "musl")]
 #[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
-#[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
+#[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
 extern "C" {}
 
 // When building with crt-static, we get `gcc_eh` from the `libc` crate, since
diff -Naur rustc-1.48.0-src.orig/src/bootstrap/dist.rs rustc-1.48.0-src/src/bootstrap/dist.rs
--- rustc-1.48.0-src.orig/src/bootstrap/dist.rs		2020-12-22 16:39:30.504249113 +0100
+++ rustc-1.48.0-src/src/bootstrap/dist.rs			2020-12-22 16:42:08.663006830 +0100
@@ -1016,7 +1016,7 @@
         copy_src_dirs(
             builder,
             &builder.src,
-            &["library", "src/llvm-project/libunwind"],
+            &["library"],
             &[
                 // not needed and contains symlinks which rustup currently
                 // chokes on when unpacking.

