--- a/src/tbb/dynamic_link.cpp
+++ b/src/tbb/dynamic_link.cpp
@@ -413,9 +413,9 @@
         int flags = RTLD_NOW;
         if (local_binding) {
             flags = flags | RTLD_LOCAL;
-#if __linux__ && !__ANDROID__ && !__TBB_USE_ADDRESS_SANITIZER
+#if __linux__ && defined(__GLIBC__) && !__TBB_USE_ADDRESS_SANITIZER
             flags = flags | RTLD_DEEPBIND;
-#endif /*__linux__ && !__ANDROID__ && !__TBB_USE_ADDRESS_SANITIZER*/
+#endif /*__linux__ && defined(__GLIBC__) && !__TBB_USE_ADDRESS_SANITIZER*/
         } else {
             flags = flags | RTLD_GLOBAL;
         }
