--- a/scripts/make.sh
+++ b/scripts/make.sh
@@ -56,7 +56,7 @@
 unset A B DOTPROG DIDNEWER
 
 # Force full rebuild if our compiler/linker options changed
-cmp -s <(compflags|sed '5,8!d') <($SED '5,8!d' "$GENDIR"/build.sh 2>/dev/null)||
+[ "$(compflags|sed '5,8!d')" != "$($SED '5,8!d' "$GENDIR"/build.sh 2>/dev/null)" ]||
   rm -rf "$GENDIR"/* # Keep symlink, delete contents
 mkdir -p "$UNSTRIPPED"  "$(dirname $OUTNAME)" || exit 1
 
