--- a/Makefile
+++ b/Makefile
@@ -7619,7 +7619,7 @@
 	@true
 
 lib/route/pktloc_syntax.c: lib/route/pktloc_syntax.y lib/route/.dirstamp
-	$(AM_V_GEN) $(YACC) -d $(YFLAGS) -o $@ $<
+	$(AM_V_GEN) $(YACC) -p pktloc_ -d $(YFLAGS) -o $@ $<
 
 lib/route/cls/ematch_grammar.h: lib/route/cls/ematch_grammar.c
 	@true
@@ -7631,7 +7631,7 @@
 	@true
 
 lib/route/cls/ematch_syntax.c: lib/route/cls/ematch_syntax.y lib/route/cls/.dirstamp
-	$(AM_V_GEN) $(YACC) -d $(YFLAGS) -o $@ $<
+	$(AM_V_GEN) $(YACC) -p ematch_ -d $(YFLAGS) -o $@ $<
 
 $(lib_libnl_route_3_la_OBJECTS): $(grammar_files_headers)
 
