--- a/configure
+++ b/configure
@@ -10,16 +10,8 @@
 EOF
     $CC -o conftest conftest.c > /dev/null 2>&1
     if [ $? -eq 0 ] ; then
+      cc="$CC"
+      return 0
-      ./conftest
-      if [ $? -eq 0 ] ; then
-	rm -f conftest conftest.c
-	cc="$CC"
-	return 0
-      else
-	echo "could not build working executables"
-	echo "Please ensure your C compiler is a native compiler"
-	exit 1
-      fi
     else
       rm -f conftest conftest.c
     fi
