--- a/cmake/Findllvm.cmake
+++ b/cmake/Findllvm.cmake
@@ -107,22 +107,8 @@
       endif()
   endfunction(NEED_TARGET)
   NEED_TARGET("AArch64")
-  NEED_TARGET("AMDGPU")
-  NEED_TARGET("ARM")
-  NEED_TARGET("AVR")
-  NEED_TARGET("BPF")
-  NEED_TARGET("Hexagon")
-  NEED_TARGET("Lanai")
-  NEED_TARGET("Mips")
-  NEED_TARGET("MSP430")
-  NEED_TARGET("NVPTX")
-  NEED_TARGET("PowerPC")
   NEED_TARGET("RISCV")
-  NEED_TARGET("Sparc")
-  NEED_TARGET("SystemZ")
-  NEED_TARGET("WebAssembly")
   NEED_TARGET("X86")
-  NEED_TARGET("XCore")
 
   if(ZIG_STATIC_LLVM)
     execute_process(
